cli-commands/verifier/verifier.js

```diff
@@ -0,0 +1,169 @@
+const { runWithoutWriteFiles } = require('../flattener/flatten');
+const axios = require('axios');
+const querystring = require('querystring');
+const logger = require('../../logger-service/logger-service').logger;
+const colors = require('../../utils/colors');
+const ethers = require('ethers');
+const DEFAULT_SEND_REQUEST_TIMEOUT = 10000;
+const DEFAULT_CHECK_STATUS_TIMEOUT = 5000;
+class Verifier {
+
+	constructor() {
+
+	}
+
+	async verifySmartContract(contractWrapper, deploymentArguments, libraries, defaultOverrides) {
+
+		this.apiKey = defaultOverrides.apiKey;
+		this.contractAddress = contractWrapper.contractAddress;
+		this.contractName = contractWrapper._contract.contractName;
+		this.flattenCode = await this._flattenSourceCode(contractWrapper);
+		this.optimization = contractWrapper._contract.compiler.optimizer;
+		this.runs = contractWrapper._contract.compiler.runs;
+		this.optimization = contractWrapper._contract.compiler.optimizer ? 1 : 0;
+		this.constructorArguments = await this._buildConstructorArguments(contractWrapper, deploymentArguments);
+		this.libraries = this._buildLibrariesArguments(libraries);
+		this.apiUrl = await this._buildApiUrl(contractWrapper.provider);
+		this.solcVersionCompiler = this._buildSolcVersionCompiler(contractWrapper._contract.compiler.version);
+		logger.log(`Attempting to verify your contract: ${colors.colorName(this.contractName)} on network ${colors.colorParams(this.name)}`);
+		let data = this._constructRequestData();
+
+		const response = await this._sendVerificationRequest(data, defaultOverrides);
+		return await this._checkVerificationStatus(response, defaultOverrides);
+	}
+
+	async _flattenSourceCode(contractWrapper) {
+		const regexp = /[0-9]?\.[0-9]?\.[0-9]?/;
+		const solcVersion = regexp.exec(contractWrapper._contract.compiler.version)[0];
+		const sourceCode = await runWithoutWriteFiles(`${this.contractName}.sol`, solcVersion);
```

Comment:
You probably can take the contract name from the contractWrapper instead of making class property